pkg/gatherers/clusterconfig/operators_pods_and_events.go

```diff
@@ -313,6 +318,35 @@ func getContainerLogs(ctx context.Context,
 	return records
 }
 
+// getLogWithStacktracing search for the first stack trace line and offset it
+func getLogWithStacktracing(logArray []string) []string {
+	var log []string
+	var found bool
+
+	for idx := range logArray {
+		line := logArray[idx]
+		// if it already found the stack we just want to keep add the log till the end
+		if found {
+			log = append(log, line)
+			continue
+		}
+		// tries to find the stack message at the given line and add it to the log
+		if found = stackTraceRegex.MatchString(line); found {
+			log = append(log, line)
+			continue
+		}
+		// if it didn't find the line yet, keep add it as offset limiting by the arbitrary
+		// offset limit value
+		log = append(log, line)
+		if len(log) > int(logLinesOffset) {
+			log[0] = "" // to avoid memory leaks
+			log = log[1:]
```

Comment:
Sorry. I don't understand this logic. You set the first line to "" and then you omit it. Why do we want to omit the first line?

Reply:
If you don't clean up the value stored at the index before you remove it, you could have a memory leak.

Reply:
First, please discuss why we have to omit the first line, and then maybe talk about a potential memory leak.

Reply:
@tremes I think the idea is that we always append lines to `log` and if we haven't found the stacktrace yet, but already have too many lines in the `log` we set the first line to an empty string and change a reference to an underlying array so that it now ignores this line, so we get something like circular buffer, but implemented in a very strange way :)

Maybe it's better to rewrite it like this:

```
find the index of the first string matching stackStaceRegex
take all the items from "this index we've just found minus logLinesOffset" till the end and copy it to a new array
```

Of course checking the corner cases like if there was nothing found. If I got the logic correctly.

Reply:
Let's discuss it tomorrow.